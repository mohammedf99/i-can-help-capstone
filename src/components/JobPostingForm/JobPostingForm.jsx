import React, { useContext, useEffect, useState } from "react";
import { Row, Col } from "antd";
import { toast } from "react-toastify";
import { JobPostingContainer } from "./JobPostingForm.styled";
import { post } from "../../Utilities/FirebaseUtilities";
import AuthContext from "../../Utilities/Contexts/AuthContext";
import DataContext from "../../Utilities/Contexts/DataContext";

const JobPostingForm = () => {
  const { currentUser } = useContext(AuthContext);
  const { userData, getPosts } = useContext(DataContext);
  const [values, setValues] = useState({
    price: null,
    type: null,
    jobType: null,
    employment: null,
    location: null,
    title: null,
    jobDescription: null,
    picture: null,
    createdDate: Date.now(),
    gender: userData?.gender || null,
  });

  useEffect(() => {
    if (userData) setValues((prev) => ({ ...prev, gender: userData?.gender }));
  }, [userData]);

  const onImageChange = (e) => {
    const reader = new FileReader();
    const file = e.target.files[0]; // get the supplied file
    // if there is a file, set image to that file
    if (file) {
      reader.onload = () => {
        if (reader.readyState === 2) {
          setValues((prev) => ({ ...prev, picture: file }));
        }
      };
      reader.readAsDataURL(e.target.files[0]);
      // if there is no file, set image back to null
    } else {
      setValues(null);
    }
  };

  const hasNull = () => {
    for (const value in values) {
      if (values[value] === null) return true;
    }

    return false;
  };

  const handlePost = () => {
    if (!hasNull()) {
      if (currentUser) post(values, currentUser.uid, () => getPosts());
    } else {
      toast.error("Please fill the Form", {
        position: "top-right",
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
      });
    }
  };

  return (
    <JobPostingContainer>
      <form action="">
        <Row>
          <input
            type="text"
            name="title"
            id="title"
            placeholder="Job title"
            onChange={(e) => {
              e.persist();
              setValues((prev) => ({
                ...prev,
                title: e.target.value,
              }));
            }}
          />
        </Row>
        <Row>
          <Col>
            <textarea
              name="post"
              id="post"
              cols="140"
              rows="3"
              placeholder="Job description..."
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  jobDescription: e.target.value,
                }));
              }}
            />
          </Col>
        </Row>
        <Row
          style={{
            display: "flex",
            flexWrap: "wrap",
          }}
        >
          <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
            <input
              type="number"
              name="salary"
              id="salary"
              placeholder="Salary in USD"
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  price: e.target.value,
                }));
              }}
            />
          </Col>
          <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
            <select
              className="post-select"
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  type: Number(e.target.value),
                }));
              }}
            >
              <option selected disabled>
                You are a(n)
              </option>
              <option value={1}>Employer</option>
              <option value={2}>Jobseeker</option>
            </select>
          </Col>
          <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
            <select
              className="post-select"
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  jobType: e.target.value,
                }));
              }}
            >
              <option selected disabled>
                Job type
              </option>
              <option value="Babysitter">Babysitter</option>
              <option value="Cleaner">Cleaner</option>
              <option value="Design">Design</option>
              <option value="Gardner">Gardner</option>
              <option value="Marketing">Marketing</option>
              <option value="Production">Production</option>
              <option value="TranslatorTranslator">Translator</option>
              <option value="Worker">Worker</option>
              <option value="open">open</option>
              <option value="Others">Others</option>
            </select>
          </Col>
          <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
            <select
              className="post-select"
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  employment: e.target.value,
                }));
              }}
            >
              <option selected disabled>
                Employment
              </option>
              <option value="Contract">Contract</option>
              <option value="Full-time">Full-time</option>
              <option value="Part-time">Part-time</option>
            </select>
          </Col>
          {values.employment === "Contract" && (
            <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
              <input
                className="post-select"
                placeholder="Time"
                onChange={(e) => {
                  e.persist();
                  setValues((prev) => ({
                    ...prev,
                    time: e.target.value,
                  }));
                }}
              />
            </Col>
          )}
          <Col style={{ flex: 1, minWidth: "120px", margin: "10px" }}>
            <select
              className="post-select"
              onChange={(e) => {
                e.persist();
                setValues((prev) => ({
                  ...prev,
                  location: e.target.value,
                }));
              }}
            >
              <option selected disabled>
                Location
              </option>
              <option value="Erbil">Erbil</option>
              <option value="Duhok">Duhok</option>
              <option value="Halabja">Halabja</option>
              <option value="Kerkuk">Kerkuk</option>
              <option value="Sulaymaniyah">Sulaymaniyah</option>
            </select>
          </Col>
        </Row>
        <Row justify="space-between" align="middle" className="btn-row">
          <input type="file" onChange={(e) => onImageChange(e)} />
          <button
            className="post-btn"
            type="button"
            onClick={() => handlePost()}
          >
            Post
          </button>
        </Row>
      </form>
    </JobPostingContainer>
  );
};

export default JobPostingForm;
