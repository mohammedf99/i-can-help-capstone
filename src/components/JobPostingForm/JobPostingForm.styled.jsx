import styled from "styled-components";

export const JobPostingContainer = styled.div`
  border: 2px solid #1c1259;
  padding: 0.5rem;
  border-radius: 5px;
  background-color: #fafaff;
  color: #1c1259;
  width: 99%;

  #title {
    border: none;
    outline: none;
    width: 300px;
    background-color: inherit;
    font-size: 1rem;
    border-bottom: 2px solid #cacaea;
    margin-bottom: 1rem;
  }

  textarea {
    resize: none;
    width: 100%;
    height: auto;
    border: none;
    outline: none;
    background-color: transparent;
    font-size: 1rem;
    font-weight: 500;
  }

  #title::placeholder,
  textarea::placeholder {
    color: #cacaea;
  }

  #salary,
  .post-select {
    font-size: 1rem;
    color: #1c1259;
    background-color: #cacaea;
    border-radius: 5px;
    padding-left: 10px;
    border: 0;
    height: 40px;
    outline: none;
    width: 95%;
  }

  #salary::placeholder {
    color: #1c1259;
  }

  .post-btn {
    color: #fafaff;
    background-color: #1c1259;
    width: 15%;
    height: 40px;
    margin-top: 1rem;
    margin-right: 1rem;
    border: none;
    border-radius: 25px;
  }

  .post-btn:hover {
    opacity: 0.8;
  }

  .upload-btn {
    background-color: #cacaea;
    color: #1c1259;
    border: 2px solid #cacaea;
    width: 185px;
    height: 40px;
  }

  .btn-row {
    margin-top: 1.5rem;
  }
`;
