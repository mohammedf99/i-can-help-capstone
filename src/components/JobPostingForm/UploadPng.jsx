import React, { useState } from "react";
import { Upload, Button, message, Tooltip } from "antd";
import { UploadOutlined } from "@ant-design/icons";

const UploadPng = () => {
  const props = {
    beforeUpload: (file) => {
      if (file.type !== "image/png") {
        message.error(`${file.name} is not a png file`);
      }
      return file.type === "image/png" ? true : Upload.LIST_IGNORE;
    },
    onChange: (info) => {
      console.log(info.fileList);
    },
  };
  return (
    <Upload {...props}>
      <Tooltip placement="bottom" title="PNG image only">
        <Button icon={<UploadOutlined />} className="upload-btn">
          Upload image
        </Button>
      </Tooltip>
    </Upload>
  );
};

export default UploadPng;
