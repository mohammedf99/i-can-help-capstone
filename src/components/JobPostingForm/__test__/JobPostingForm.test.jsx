import * as React from "react";
import * as renderer from "react-test-renderer";
import JobPostingForm from "../JobPostingForm";

it("renders Navbar correctly", () => {
  const tree = renderer.create(<JobPostingForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
