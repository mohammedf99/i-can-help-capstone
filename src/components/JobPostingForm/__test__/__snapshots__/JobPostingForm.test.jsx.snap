// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div
  className="JobPostingFormstyled__JobPostingContainer-sc-1bu3b4i-0 bHIcoe"
>
  <form
    action=""
  >
    <div
      className="ant-row"
      style={Object {}}
    >
      <input
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="jobTitle"
        type="text"
      />
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <textarea
          cols="140"
          id="post"
          name="post"
          onChange={[Function]}
          placeholder="jobDesc"
          rows="3"
        />
      </div>
    </div>
    <div
      className="ant-row"
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <div
        className="ant-col"
        style={
          Object {
            "flex": 1,
            "margin": "10px",
            "minWidth": "120px",
          }
        }
      >
        <input
          id="salary"
          name="salary"
          onChange={[Function]}
          placeholder="salaryInUsd"
          type="number"
        />
      </div>
      <div
        className="ant-col"
        style={
          Object {
            "flex": 1,
            "margin": "10px",
            "minWidth": "120px",
          }
        }
      >
        <select
          className="post-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            youAreAn
          </option>
          <option
            value={1}
          >
            Employer
          </option>
          <option
            value={2}
          >
            Jobseeker
          </option>
        </select>
      </div>
      <div
        className="ant-col"
        style={
          Object {
            "flex": 1,
            "margin": "10px",
            "minWidth": "120px",
          }
        }
      >
        <select
          className="post-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            jobType
          </option>
          <option
            value="Babysitter"
          >
            Babysitter
          </option>
          <option
            value="Cleaner"
          >
            Cleaner
          </option>
          <option
            value="Design"
          >
            Design
          </option>
          <option
            value="Gardner"
          >
            Gardner
          </option>
          <option
            value="Marketing"
          >
            Marketing
          </option>
          <option
            value="Production"
          >
            Production
          </option>
          <option
            value="TranslatorTranslator"
          >
            Translator
          </option>
          <option
            value="Worker"
          >
            Worker
          </option>
          <option
            value="open"
          >
            open
          </option>
          <option
            value="Others"
          >
            Others
          </option>
        </select>
      </div>
      <div
        className="ant-col"
        style={
          Object {
            "flex": 1,
            "margin": "10px",
            "minWidth": "120px",
          }
        }
      >
        <select
          className="post-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            employment
          </option>
          <option
            value="Contract"
          >
            Contract
          </option>
          <option
            value="Full-time"
          >
            Full-time
          </option>
          <option
            value="Part-time"
          >
            Part-time
          </option>
        </select>
      </div>
      <div
        className="ant-col"
        style={
          Object {
            "flex": 1,
            "margin": "10px",
            "minWidth": "120px",
          }
        }
      >
        <select
          className="post-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            location
          </option>
          <option
            value="Erbil"
          >
            Erbil
          </option>
          <option
            value="Duhok"
          >
            Duhok
          </option>
          <option
            value="Halabja"
          >
            Halabja
          </option>
          <option
            value="Kerkuk"
          >
            Kerkuk
          </option>
          <option
            value="Sulaymaniyah"
          >
            Sulaymaniyah
          </option>
        </select>
      </div>
    </div>
    <div
      className="ant-row ant-row-space-between ant-row-middle btn-row"
      style={Object {}}
    >
      <input
        onChange={[Function]}
        type="file"
      />
      <button
        className="post-btn"
        onClick={[Function]}
        type="button"
      >
        post
      </button>
    </div>
  </form>
</div>
`;
